jsx/StaticDataTable.js

```diff
@@ -295,6 +295,68 @@ class StaticDataTable extends Component {
     });
   }
 
+  /**
+   * Get if the current sorting column has mixed types in its values.
+   *
+   * @return {bool} true if mixed types, else false.
+   */
+  hasMixedTypes() {
+    // TODO: data column type check should probably be done once at init,
+    // meaning when receiving data, to find which columns have mixed types.
+    let typesFound = null;
+
+    // not the default column
+    if (this.state.SortColumn === -1) {
+      return false;
+    }
+
+    // is the current sorted column with mixed type?
+    let isMixedType = false;
+
+    // Only checks string or number types, others are considered undefined.
+    // Break out of this loop once we encounter two mixed types:
+    // number and string inside the sorted column.
+    for (const row of this.props.Data) {
+      // cell value
+      let val = row[this.state.SortColumn];
+
+      // if null or undefined, go to the next iteration
+      if (val == null) {
+        continue;
+      }
+
+      // check number
+      if (!isNaN(val) && typeof val !== 'object') {
```

Comment:
This check can flag empty strings as numbers (among others) :
```ts
!isNaN("") && typeof "" !== "object" // true
```

This should actually still work (although it is a little hacky), but after giving it some thought, I don't think there is really any good way to detect variable types if modules send numbers as strings instead of proper numbers unfortunately (note that this is an argument in favor of typing in general).